### DISCUSSION TITLE
double-clicking a PDF re-opens existing Sumatrapdf windows

### DISCUSSION THREAD

**Discussion by xerces8** [sumatrapdfreader/sumatrapdf#3030] [Q&A]
I have v3.4.6 installed on Windows 10 Pro 21H2.

If I have SumatraPDF running with 4 PDF documents open and then doubleclick a PDF file in Explorer, then the current 4 documents are opened again and the new clicked PDF is finally opened. So now I have 9 documents open (in SumatraPDF). If I repeat (doubleclick another PDF file) then I get 19 open documents.

Is this a bug? Misconfiguration? Anyone else having this?

**Comment by mathsmatics** [comment]
it looks like a bug,i don't have it.

**Comment by xerces8** [comment]
Maybe this is relevant, I think this is what happened:

 - open several PDF files in SumatraPDF
 - reboot
 - SumatraPDF  is automatically opened with the same PDF files as before reboot
 - doubleclick a PDF file in Explorer -> the described issue happens

I will check next time I reboot. Currently, I started a "clean" copy of SumatraPDF  and don't get the issue, if I open another PDF, it is opened, without reopening the other already open PDF files.

**Reply by GitHubRulesOK** [reply]
hmmm there was a lot of complaints SumatraPDF "lost track" of opened files during dirty windows closure on windows updates where especially home users could not cleanly close before shutdown/reboot.

There were changes made to try to maintain history recovery in several similar cases, so it seems that's having an effect in this case

**Comment by irmlab** [comment]
The best solution would be to let users have "sumatra-session.ssf" files, with the list of opened files (and/or some other options)
User could then save and open session on local folder, in the folder hierarchy or use the "global" session of portable.

The idea is as follow, The file menu will include 2 options:
-- "Save session" item option to save the opened files 
-- "Open session" item to reopen a session.

When Sumatra is started without a file parameter, it looks for a Sumatra-session file in the working folder then upwards in the directory hierarchy. This would work for portable or installed version.

Sumatra is a very nice SW, but the file history management is a bit inconsistent, I agree :)

**Comment by GitHubRulesOK** [comment]
SumatraPDF both portable and Single User both look for 2 settings files in Current.exe Working Directory, Only Multi-User install looks sideways into users local appdata directory.

Sessions can be saved separately (Notepad Ctrl+S) for different histories (but not jointly) in old forum I showed you can add a backup for any day hour or minute on demand but the management of backups and user -appdata choices is per each user needs and preferences.

skip  to end of https://forum.sumatrapdfreader.org/t/open-files-from-multiple-tabs-in-acrobat-or-other-app/3431 it is not that difficult to run  if you are cautious about the necessary double quotes.

**Reply by irmlab** [reply]
I just opened an "Idea" discussion :) 
I noticed what you explain, but it is not well managed 
And you explain why :  because these .INI files, depend on system or software configuration.
Instead, Session files are simple text files with no dependency.

Yes:  a lot of Editors use similar sessions files. 
My inspiration is from Scite Editor, (look for "Scintilla", "Scite" editor, Neil Hodgson
Here: https://www.scintilla.org/

Let's continue in the "Idea" topic discussion.  

BR!
